### ISSUE TITLE
Bitwise left shift assignment

### ISSUE THREAD

**Issue by DCGroothuizenDijkema** [microsoft/vscode-cpptools#4512]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 Pro 1809
- VS Code Version: 1.39.2
- C/C++ Extension Version: 0.26.1
- Other extensions you installed (and if the issue persists after disabling them): Issue persists after all other extensions are disabled.
- A clear and concise description of what the bug is: defining the bitwise leftshfit assignment operator for a class removes syntax highlighting of keywords.

**To Reproduce**

1. Create a new document, select the language to be "C++", and type:
```
template <typename T>
class DummyContainer
{
public:
  DummyContainer(unsigned int size) : size(size)
  {
    container=new T[size];
  }

  ~DummyContainer()
  {
    delete [] container;
  }

  DummyContainer operator>>=(unsigned int n)
  {
    for (int itr=0;itr+n<size;++itr)
    {
      container[itr]=container[itr+n];
    }
    return *this;
  }

  DummyContainer operator<<=(unsigned int n)
  {
    for (int itr=0;itr+n<size;++itr)
    {
      container[itr]=container[itr+n];
    }
    return *this;
  }

private:
  T *container;
  int size;
};
```

**Expected behavior**
`for`, `return`, `this`, and `private` should highlight purple, like they do previously. Note that the bitwise right shift assignment operator does not cause this issue.

**Screenshots**
![bitwise_left_shift_assignment](https://user-images.githubusercontent.com/41760053/67824638-da87a900-fb2b-11e9-8d47-70c9da9ce05f.png)

**Comment by Colengms** [comment]
Hi @DCGroothuizenDijkema .  There are 2 types of colorization occurring when using the C/C++ Extension: syntactic and semantic.  Syntactic colorization is applied by VS Code itself, based on TextMate grammar for the language.  Language keywords are colored this way.  The C/C++ Extension adds semantic colorization as a layer over syntactic colorization.  Semantic colorization colors user-defined symbols that require a partial compilation to recognize, such as class names, variable names, namespace names.

One way to determine if the issue is syntactic or semantic is to disable semantic colorization by setting ```C_Cpp.enhancedColorization``` to ```Disabled```.  In your case, it looks like the issue is syntactic (as it involved language keywords, such as ```for``` and ```return```).  @Jeff-Hykin contributes TextMate grammar for C/C++ to VS Code.  His repo for TextMate grammar is here: https://github.com/jeff-hykin/cpp-textmate-grammar .  If he doesn't respond directly to this issue, I'd suggest opening the issue in his repo.

**Comment by jeff-hykin** [comment]
@DCGroothuizenDijkema you can install the [Better C++ Syntax](https://marketplace.visualstudio.com/items?itemName=jeff-hykin.better-cpp-syntax&ssr=false#review-details) to get the fix that was just pushed just now with `v1.14.11`.

Thanks for pointing this out, this is caused by a more general ordering issue that we were unaware of (`<` came before `<<=`). Good news is, its not to hard to fix and it will prevent/fix several hidden bugs with other operators and keywords.

**Comment by DCGroothuizenDijkema** [comment]
Brilliant: all fixed. @jeff-hykin and @Colengms, thank you both very much.
